sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVector.java

```diff
@@ -57,6 +60,25 @@ public static RapidsHostColumnVector[] extractColumns(ColumnarBatch batch) {
     return vectors;
   }
 
+  public static ColumnarBatch incRefCounts(ColumnarBatch batch) {
+    for (RapidsHostColumnVector rapidsHostCv: extractColumns(batch)) {
+      rapidsHostCv.incRefCount();
+    }
+    return batch;
+  }
+
+  public static long getTotalHostMemoryUsed(ColumnarBatch batch) {
+    long sum = 0;
+    if (batch.numCols() > 0) {
+      HashSet<RapidsHostColumnVector> found = new HashSet<>();
+      for (RapidsHostColumnVector rapidsHostCv: extractColumns(batch)) {
+        if (found.add(rapidsHostCv)) {
```

Comment:
Can you comment why are we deduping after extractColumns?

Reply:
It's possible that we could construct a ColumnarBatch with columns that are exactly the same, via a no-op projection Something like: `select foo, foo, foo, bar from table`. I'd expect a project like this to produce a ColumnarBatch with 4 columns, where the instance column vector `foo` is in the columns array at positions 0, 1, and 2, and having a reference count of 3. `bar` has a refcount of 1. When we close this batch, `bar` goes from 1 to 0, and `foo` needs to go from 3->0 via each of the ref count decrements for the column slots.

Reply:
so without `hashCode` and `equals` we rely on reference identity

Reply:
Correct. HostColumnVectors own their data, so if we have two different references its expected that they point to two different cuDF native structures. I am not sure we need to implement this unless we aliased the columns using views, right now that shouldn't be the case.

Reply:
I did not say we need to implement it, I am just trying to figure what and why is being done here.

Reply:
I missed the part about adding this as a comment, I've added that to this follow on issue: https://github.com/NVIDIA/spark-rapids/issues/9121